src/components/ListItem.tsx

```diff
@@ -1,8 +1,70 @@
-import * as api from '../api';
-import './ListItem.css';
+import "./ListItem.css";
+import { updateItem, ListItem } from "../api";
+import { useEffect, useState } from "react";
+import toast from "react-hot-toast";
+import { is24HoursLater } from "../utils";
 
-type Props = Pick<api.ListItem, 'name'>;
+interface Props {
+	item: ListItem;
+	listPath: string | null;
+}
+
+export function ListItemCheckBox({ item, listPath }: Props) {
+	const [checked, setChecked] = useState(false);
+
+	const purchaseDate = item.dateLastPurchased
+		? item.dateLastPurchased.toDate()
+		: item.dateLastPurchased;
+
+	useEffect(() => {
+		if (purchaseDate) {
+			setChecked(!is24HoursLater(purchaseDate));
+		} else {
+			setChecked(false);
+		}
+	}, [item.dateLastPurchased]);
+
+	const handleCheckChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+		const isChecked = e.target.checked;
+
+		if (!isChecked && purchaseDate && !is24HoursLater(purchaseDate)) {
+			toast.error(
+				`${item.name} has already been marked as purchased in the last 24 hours.`,
+			);
+			return;
+		}
+
+		setChecked(isChecked);
+
+		if (!listPath) {
+			toast.error("Error: listPath is missing or invalid.");
+			return;
+		}
+
+		await toast.promise(updateItem(listPath, item), {
+			loading: `Marking ${item.name} as purchased!`,
+			success: `${item.name} is now marked as purchased!`,
+			error: () => {
+				setChecked(!isChecked);
+				return `${item.name} failed to add to your list of purchases. Please try again!`;
+			},
+		});
+	};
 
-export function ListItem({ name }: Props) {
-	return <li className="ListItem">{name}</li>;
+	return (
+		<div className="ListItem">
+			<label htmlFor={`checkbox-${item.id}`}>
+				<input
+					type="checkbox"
+					id={`checkbox-${item.id}`}
+					aria-label={`Mark ${item.name} as purchased.`}
+					value={item.id}
+					checked={checked}
+					onChange={handleCheckChange}
+					aria-checked={checked}
+				/>
```

Comment:
Suggestion: To improve user experience, the checkbox could be disabled while the item is within the 24-hour period. This would give a clear visual indication that users can't uncheck the item before 24 hours have passed.
You can add it as a property in the input label
`disabled={!is24HoursLater(purchaseDate)} // Disable if 24 hours haven't passed`

Reply:
yessss!!! This was definitely in sync with some of the conversations we were having. Thanks for the validation and provided code! I like this better than the error message we sent telling them "they already purchased it in 24 hours". While the goals of both cases are the same, its less jarring to not see an error message thrown to the user.